packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -189,4 +190,65 @@ export class MigrationDataSource {
   get configFileNameAfterMigration () {
     return this.ctx.lifecycleManager.legacyConfigFile.replace('.json', `.config.${this.ctx.lifecycleManager.fileExtensionToUse}`)
   }
+
+  private checkAndUpdateDuplicatedSpecs (specs: MigrationFile[]) {
+    const updatedSpecs: MigrationFile[] = []
+
+    const sortedSpecs = this.sortSpecsByExtension(specs)
+
+    sortedSpecs.forEach((spec) => {
```

Comment:
Been thinking about this function for a few days. I still think it's a good candidate for a unit test, the system tests are not exactly the best for debugging complex logic.

I think we could have done this with `reduce` and a `Set` to see if a spec already has been updated instead of the nested `_.find` in each loop, it'd be a little more efficient. This is okay though, but good to make small optimizations here and there.

Neither of these is a blocker on merging, though.

:heavy_check_mark: